### PR TITLE
fix: update nonce generation to use cryptographic RNG

### PR DESCRIPTION
**Pull request by kdenhartog**
Small change to move the example implementation to a 96 bit nonce generated by a CSPRNG

### CODE REVIEW

**Comment by w4ll3** [comment]
Thanks for the PR! Looking good, do you think you could refactor it to be part of the core library as a new function? `generateNonce()` that way we could use it to generate nonces at the backend and in the library itself (check `client.ts:117`)

**Comment by kdenhartog** [comment]
@w4ll3 I was able to get this updated just now, but I'd consider this still a work in progress at this point because I saw that the `client.test.ts file` is failing one test locally because I wasn't able to figure out how to get the examples folder to consume the new function. In theory the code works, but for whatever reason the notepad example isn't picking up the changes made to the core library. Any suggestions on this?

**Comment by w4ll3** [comment]
> @w4ll3 I was able to get this updated just now, but I'd consider this still a work in progress at this point because I saw that the `client.test.ts file` is failing one test locally because I wasn't able to figure out how to get the examples folder to consume the new function. In theory the code works, but for whatever reason the notepad example isn't picking up the changes made to the core library. Any suggestions on this?

Mmmm, that's weird. Works for me. Maybe try deleting `node_modules` (from both lib and example) and rebuild?

**Comment by kdenhartog** [comment]
Ok seems like the server is working fine now, but still have a test failure occurring for validationPositive case. Do I need to regenerate the test vectors or anything like that since the nonce is being changed?

**Comment by w4ll3** [comment]
> Ok seems like the server is working fine now, but still have a test failure occurring for validationPositive case. Do I need to regenerate the test vectors or anything like that since the nonce is being changed?

Actually the test was failing for others reasons, it should be fine now.

File: lib/client.ts [hunk 1]
```diff
@@ -283,3 +285,18 @@ export const checkContractWalletSignature = async (
...
+ * 
+ * 96 bits has been chosen as a number to sufficiently balance size and security considerations
+ * relative to the lifespan of it's usage.
+ * 
+ * @returns cryptographically generated random nonce with 96 bits of entropy encoded with
+ * an alphanumeric character set.
+ */
+export const generateNonce = (): string => {
+	return randomStringForEntropy(96);
+}
```
Comment (w4ll3) [inline lib/client.ts:302]: Please add a blank new line at the end

Comment (w4ll3) [inline lib/client.ts:302]:
```suggestion
}

```

**Review by w4ll3 — CHANGES_REQUESTED** [review]

**Review by w4ll3 — APPROVED** [review]

**Review by wyc — APPROVED** [review]

**Pull request merged**